File: FixFlow/FixFlow.Application/DTOs/Response/UserResponse.cs
using FixFlow.Core.Enums;

namespace FixFlow.Application.DTOs.Response;

public class UserResponse
{
    public int Id { get; set; }
    public string Email { get; set; } = string.Empty;
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public UserRole Role { get; set; }
    public string? ProfileImage { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: packages/fixflow_core/lib/models/responses/user_response.dart
import 'package:fixflow_core/enums/user_role.dart';

class UserResponse {
  final int id;
  final String email;
  final String firstName;
  final String lastName;
  final String? phone;
  final UserRole role;
  final String? profileImage;
  final DateTime createdAt;

  UserResponse({
    required this.id,
    required this.email,
    required this.firstName,
    required this.lastName,
    this.phone,
    required this.role,
    this.profileImage,
    required this.createdAt,
  });

  factory UserResponse.fromJson(Map<String, dynamic> json) {
    return UserResponse(
      id: json['id'] as int,
      email: json['email'] as String,
      firstName: json['firstName'] as String,
      lastName: json['lastName'] as String,
      phone: json['phone'] as String?,
      role: UserRole.fromJson(json['role']),
      profileImage: json['profileImage'] as String?,
      createdAt: DateTime.parse(json['createdAt'] as String).toLocal(),
    );
  }
}
